packages/plugin-board/src/board.js

```diff
@@ -408,6 +408,66 @@ const Board = SparkPlugin.extend({
       .then((res) => res.body);
   },
 
+  /**
+   * Registers with Mercury for sharing web socket
+   * @memberof Board.BoardService
+   * @param  {Board~Channel} channel
+   * @returns {Promise<Board~Registration>}
+   */
+  registerToShareMercury(channel) {
+    return this.spark.feature.getFeature(`developer`, `web-shared-mercury`)
+      .then((isSharingMercuryFeatureEnabled) => {
+        if (!this.spark.mercury.localClusterServiceUrls) {
+          return Promise.reject(new Error(`\`localClusterServiceUrls\` is not defined, make sure mercury is connected`));
+        }
+        else if (!isSharingMercuryFeatureEnabled) {
+          return Promise.reject(new Error(`\`web-shared-mercury\` is not enabled`));
+        }
+
+        const webSocketUrl = this.spark.device.webSocketUrl;
+        const mercuryConnectionServiceClusterUrl = this.spark.mercury.localClusterServiceUrls.mercuryConnectionServiceClusterUrl;
+
+        const data = {
+          mercuryConnectionServiceClusterUrl,
+          webSocketUrl,
+          action: `REPLACE`
+        };
+
+        return data;
+      })
```

Comment:
I don't think there's a need for this then block. can't you just call `spark.request` with `data`?